### PR TITLE
feat: generate compile_commands.json with ninja

### PR DESCRIPTION
**Pull request by legendecas**
ninja supports generating `compile_commands.json` from ninja files. The generated `compile_commands.json` resolves the generated product path as well, which is not supported in the [`gyp.generator.compile_commands_json`](https://github.com/nodejs/gyp-next/blob/main/pylib/gyp/generator/compile_commands_json.py#L70).

To generate `compile_commands.json` with ninja, define generator flag `-Gcompile_commands` when running gyp.

### CODE REVIEW

File: pylib/gyp/generator/ninja_test.py [hunk 1]
```diff
@@ -50,6 +51,16 @@ def test_BinaryNamesLinux(self):
...
 
+    def test_GenerateCompileDBWithNinja(self):
+        this_dir = os.path.abspath(os.path.dirname(__file__))
+        build_dir = os.path.normpath(os.path.join(this_dir, "../../../data/ninja"))
+        compile_db = ninja.GenerateCompileDBWithNinja(build_dir)
+        self.assertEqual(len(compile_db), 1)
+        self.assertEqual(compile_db[0]["directory"], build_dir)
+        self.assertEqual(compile_db[0]["command"], "cc my.in my.out")
+        self.assertEqual(compile_db[0]["file"], "my.in")
+        self.assertEqual(compile_db[0]["output"], "my.out")
```
[the remark this answers is not in the document]
Comment (legendecas) [reply pylib/gyp/generator/ninja_test.py:62]: I may be missing something but when there is no `assert` keyword or `self.assertEqual`, pytest doesn't report any failure when there is any mismatches.

File: pylib/gyp/generator/ninja_test.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 """ Unit tests for the ninja.py file. """
 
+import os
```
[the remark this answers is not in the document]
Comment (legendecas) [reply pylib/gyp/generator/ninja_test.py]: thanks for suggestions! updated

File: pylib/gyp/generator/ninja_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cclauss) [reply pylib/gyp/generator/ninja_test.py]: You are correct!  My oversight!

[the remark this answers is not in the document]
Comment (legendecas) [reply pylib/gyp/generator/ninja_test.py]: Updated. PTAL, thanks!

**Comment by legendecas** [comment]
@cclauss would you mind taking a look again? thank you!

**Comment by legendecas** [comment]
@cclauss kindly ping, would you mind taking another look at this? Thanks!

**Pull request merged**